File: python/q23.py
#!/usr/bin/python3

def divisors(n):
    lst = [1]
    for i in range(2,int(n**0.5)):
        if n%i==0:
            lst.append(i)
            lst.append(n/i)
    if n**0.5==int(n**0.5):
        lst.append(n**0.5)
    return lst

def isAbundant(n):
    divs = divisors(n)
    total = sum(divs)
    if total>n:
        return True
    return False

abundantNumbers = []
for i in range(12,28123+1):
    if isAbundant(i):
        abundantNumbers.append(i)

print(len(abundantNumbers))

abundantSum = []
for num1 in abundantNumbers:
    for num2 in abundantNumbers:
        abundantSum.append(num1+num2)

total = 0
for i in range(12,28123+1):
    if not i in abundantSum:
        total += i

print(total)

File: go/q23.go
package main
import ("fmt";"math")

func divisors(n int) []int{
	lst := []int{1}
	upper := int(math.Sqrt(float64(n)))
	for i:=2; i<=upper; i++{
		if n%i==0{
			lst = append(lst, i)
			lst = append(lst,n/i)
		}
	}
	if int(math.Sqrt(float64(n)))==n{
		lst = append(lst,int(math.Sqrt(float64(n))))
	}
	return lst
}

func isAbundant(n int) bool{
	divs := divisors(n)
	total := 0
	for i := 0; i<len(divs); i++{
		total += divs[i]
	}
	if total>n{
		return true
	}
	return false
}

func containedIn(i int, abundantSum []int) bool{
	for k:=0; k<len(abundantSum); k++{
		if abundantSum[k]==i{
			return true
		}
	}
	return false
}

func main() {
	fmt.Println("Program Start")
	abundantNumbers := []int{}
	for i:=12; i<=28123; i++{
		if isAbundant(i)==true{
			abundantNumbers = append(abundantNumbers,i)
		}
	}
	fmt.Println("abundantNumbers")
	abundantSum := []int{}
	for i:=0; i<len(abundantNumbers); i++{
		for j:=0; j<len(abundantNumbers); j++{
			abundantSum = append(abundantSum, abundantNumbers[i]+abundantNumbers[j])
		}
	}
	fmt.Println("abundantNumbers")
	fmt.Println(len(abundantNumbers))
	total := 0
	for i:=12; i<=28123; i++{
		if containedIn(i,abundantSum)==false{
			total += i
		}
	}
	fmt.Println(total)
}
